test/Microsoft.AspNetCore.Sockets.Common.Tests/Internal/Formatters/ServerSentEventsParserTests.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Text;
+using Microsoft.AspNetCore.Sockets.Internal.Formatters;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Sockets.Common.Tests.Internal.Formatters
+{
+    public class ServerSentEventsParserTests
+    {
+        [Theory]
+        [InlineData("data: T\r\ndata: Hello, World\r\n\r\n", "Hello, World")]
```

Comment:
Add cases for standalone `\r` in data
Add a case where the message is "data: data: \r\n"